File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/C/s826491904_commented.c
#include <stdio.h>

void solve() {
    int N, max, min, in;
    scanf("%d", &N);
    scanf("%d", &max);
    min = max;
    for (int i = 1; i < N; i++) {
        scanf("%d", &in);
        max = (max > in) ? max : in;
        min = (min < in) ? min : in;
    }
    printf("%d\n", max - min);
}

int main() {
    solve();
    return 0; // End of Code
}

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s826491904_commented.py
import sys

class Main:
    def solve(self):
        N = int(input())
        max_val = min_val = int(input())
        for _ in range(1, N):
            num = int(input())
            max_val = max(max_val, num)
            min_val = min(min_val, num)
        print(max_val - min_val)

if __name__ == "__main__":
    main = Main()
    main.solve()  # End of Code
